src/frequenz/sdk/actor/power_distributing/_battery_status.py

```diff
@@ -352,6 +354,25 @@ def _get_current_status(self) -> Status:
 
         return Status.WORKING
 
+    def _capacity_not_nan(self, msg: BatteryData) -> bool:
+        """Battery capacity should not be guessed.
+
+        If capacity is NaN, then mark battery as broken.
+
+        Args:
+            msg: battery message
+
+        Returns:
+            True if battery capacity is NaN, false otherwise.
+        """
+        if math.isnan(msg.capacity) and self._last_status == Status.WORKING:
+            _logger.warning(
+                "Battery %d capacity is NaN",
+                msg.component_id,
+            )
+            return False
```

Comment:
According to the documentation this should return False regardless of the status so I wonder if either the code or the documentation is not  right

Reply:
Done, thanks @daniel-zullo-frequenz  and @tiyash-basu-frequenz